guides/Getting Started.md

```diff
@@ -0,0 +1,696 @@
+```
+
+This code will create three new people in our database, Ryan, John and Jane. Note here that we could've used a changeset to validate the data going into the database, but the choice was made not to use one.
+
+We'll be querying for these people in this section. Let's jump in!
+
+### Fetching a single record
+
+Let's start off with fetching just one record from our `people` table:
+
+```elixir
+Friends.Person |> Ecto.Query.first
+```
+
+That code will generate an `Ecto.Query`, which will be this:
+
+```
+#Ecto.Query<from p in Friends.Person, order_by: [asc: p.id], limit: 1>
+```
+
+The code between the angle brackets `<...>` here shows the Ecto query which has been constructed. We could construct this query ourselves with almost exactly the same syntax:
+
+```elixir
+require Ecto.Query
+Ecto.Query.from p in Friends.Person, order_by: [asc: p.id], limit: 1
+```
+
+We need to `require Ecto.Query` here so that the module is available for us to use. Then it's a matter of calling the `from` function from `Ecto.Query` and passing in the code from between the angle brackets. As we can see here, `Ecto.Query.first` saves us from having to specify the `order` and `limit` for the query.
+
+To execute the query that we've just constructed, we can call `Friends.Repo.one`:
+
+```elixir
+Friends.Person |> Ecto.Query.first |> Friends.Repo.one
+```
+
+The `one` function retrieves just one record from our database and returns a new struct from the `Friends.Person` module:
+
+```elixir
+%Friends.Person{__meta__: #Ecto.Schema.Metadata<:loaded>, age: 28,
+ first_name: "Ryan", id: 1, last_name: "Bigg"}
+```
+
+Similar to `first`, there is also `last`:
+
+```elixir
+Friends.Person |> Ecto.Query.last |> Friends.Repo.one
+%Friends.Person{__meta__: #Ecto.Schema.Metadata<:loaded>, age: 26,
+ first_name: "Jane", id: 3, last_name: "Smith"}
+ ```
+
+The `Ecto.Repo.one` function will only return a struct if there is one record in the
+result from the database. If there is more than one record returned, an
+`Ecto.MultipleResultsError` exception will be thrown. Some code that would
+cause that issue to happen is:
+
+```elixir
+Friends.Person |> Friends.Repo.one
+```
+
+We've left out the `Ecto.Query.first` here, and so there is no `limit` or `order` clause applied to the executed query. We'll see the executed query in the debug log:
+
+```
+[timestamp] [debug] SELECT p0."id", p0."first_name", p0."last_name", p0."age" FROM "people" AS p0 [] OK query=1.8ms
+```
+
+Then immediately after that, we will see the `Ecto.MultipleResultsError` exception:
+
+```
+** (Ecto.MultipleResultsError) expected at most one result but got 3 in query:
+
+from p in Friends.Person
+
+    lib/ecto/repo/queryable.ex:67: Ecto.Repo.Queryable.one/4
+```
+
+This happens because Ecto doesn't know what one record out of all the records
+returned that we want. Ecto will only return a result if we are explicit in
+our querying about which result we want.
```

Comment:
It might be good idea to mention here that when no record is returned `Repo.one` will return `nil`.